Improve static types migration docs (#6544)
nextflow-io/nextflow: docs/reference/operator.md

As shown in the above example, deeply nested collections are also flattened.

:::{tip}

> As a best practice, it is recommended to use [`flatMap`](#flatmap) instead of `flatten`, since `flatMap` only flattens one level and has a well-defined return type.

Reviewer: ```suggestion
Use [`flatMap`](#flatmap) instead of `flatten` to apply a mapping function. `flatMap` only flattens one level and has a well-defined return type.
```
I don't think I captured this properly. But it would be good to explain why this is recommended.

`remainder`
: When `true`, incomplete tuples are emitted with `null` values for missing elements, otherwise they are discarded (default: `false`). 

:::{tip}

> As a best practice, it is recommended to use [`flatMap`](#flatmap) instead of `transpose`, since `flatMap` has a well-defined return type.

Reviewer: ```suggestion
As `flatMap` has a well-defined return type, it is recommended to use [`flatMap`](#flatmap) instead of `transpose`.
```
